substrate-node/pallets/pallet-smart-contract/src/migrations/v12.rs

```diff
@@ -129,42 +131,47 @@ pub fn migrate_to_version_12<T: Config>() -> frame_support::weights::Weight {
     T::DbWeight::get().reads_writes(total_reads, total_writes)
 }
 
-fn migrate_contract_lock_to_contract_payment_state<T: Config>(contract_id: u64) -> (u64, u64) {
+fn migrate_contract_lock_to_contract_payment_state<T: Config>() -> (u64, u64) {
     let mut reads = 0;
     let mut writes = 0;
 
-    let old_contract_lock = ContractLock::<T>::take(contract_id);
-    reads.saturating_inc();
-    writes.saturating_inc();
-
-    ContractPaymentState::<T>::insert(
-        contract_id,
-        super::types::v12::ContractPaymentState {
-            standard_reserve: BalanceOf::<T>::zero(),
-            additional_reserve: BalanceOf::<T>::zero(),
-            standard_overdraft: old_contract_lock.amount_locked,
-            additional_overdraft: old_contract_lock.extra_amount_locked,
-            last_updated_seconds: old_contract_lock.lock_updated,
-            cycles: old_contract_lock.cycles,
-        },
-    );
-    writes.saturating_inc();
+    for (contract_id, old_contract_lock) in ContractLock::<T>::drain() {
+        reads.saturating_inc();
+        writes.saturating_inc();
+
+        ContractPaymentState::<T>::insert(
+            contract_id,
+            super::types::v12::ContractPaymentState {
+                standard_reserve: BalanceOf::<T>::zero(),
+                additional_reserve: BalanceOf::<T>::zero(),
+                standard_overdraft: old_contract_lock.amount_locked,
+                additional_overdraft: old_contract_lock.extra_amount_locked,
+                last_updated_seconds: old_contract_lock.lock_updated,
+                cycles: old_contract_lock.cycles,
+            },
+        );
+        reads.saturating_inc();
+    };
 
     (reads, writes)
 }
 
-fn remove_all_balances_locks<T: Config>(account_id: &T::AccountId) -> (u64, u64) {
+fn remove_all_balances_locks<T: Config>() -> (u64, u64) {
     let mut reads = 0;
     let mut writes = 0;
-
-    let locks = pallet_balances::Pallet::<T>::locks(account_id);
-    reads.saturating_inc();
-
-    for lock in locks {
-        log::debug!("Removing lock: {:?} for account: {:?}", lock.id, account_id);
-        pallet_balances::Pallet::<T>::remove_lock(lock.id, account_id);
+    // Get only the accounts with locks
+    for (account_id, _) in pallet_balances::Locks::<T>::iter() {
         reads.saturating_inc();
-        writes.saturating_inc();
+        // Fetch all locks for the account
+        let locks = pallet_balances::Pallet::<T>::locks(&account_id);
+        reads.saturating_inc();
+
+        // Remove each lock
+        for lock in locks {
+            pallet_balances::Pallet::<T>::remove_lock(lock.id, &account_id);
+            reads.saturating_inc();
```

Comment:
do we need this `reads.saturating_inc();` here?

Reply:
AFAIK, removing an item from the storage involves one read (check if key exists) and one write operation (remove corresponding entry), no?

Reply:
I know it's mainly a write operation, but there is a potential for a read operation depending on how the inner functionality is implemented on the client's database.
It's safer to take into account these extra costs unless we are completely certain.